src/Simplify.cpp

```diff
@@ -1436,6 +1433,98 @@ class Simplify : public IRMutator {
+                   is_simple_const(min_b->b)) {
+            // min(x, c1) - min(x, c2) where c1 and c2 are constants
+            // if c1 >= c2 -> clamp(x, c2, c1) - c2
+            // else -> c1 - clamp(x, c1, c2)
+            if (is_one(mutate(min_a->b >= min_b->b))) {
+                expr = mutate(clamp(min_a->a, min_b->b, min_a->b) - min_b->b);
+            } else {
+                expr = mutate(min_a->b - clamp(min_a->a, min_a->b, min_b->b));
+            }
+        } else if (no_overflow(op->type) &&
+                   max_a &&
+                   max_b &&
+                   equal(max_a->a, max_b->a) &&
+                   is_simple_const(max_a->b) &&
+                   is_simple_const(max_b->b)) {
+            // max(x, c1) - max(x, c2) where c1 and c2 are constants
+            // if c1 >= c2 -> c1 - clamp(x, c2, c1)
+            // else -> clamp(x, c1, c2) - c2
+            if (is_one(mutate(max_a->b >= max_b->b))) {
+                expr = mutate(max_a->b - clamp(max_a->a, max_b->b, max_a->b));
+            } else {
+                expr = mutate(clamp(max_a->a, max_a->b, max_b->b)- max_b->b);
+            }
+        } else if (no_overflow(op->type) &&
+                   min_a &&
+                   min_b &&
+                   is_simple_const(mutate(min_a->a - min_b->b)) &&
+                   is_simple_const(mutate(min_a->b - min_b->a))) {
+            // Canonicalize min(a + c1, b + c2) - min(b + c4, a + c3)
+            //     where c1, c2, c3, and c4 are constants
+            // into min(a + c1, b + c2) - min(a + c3, b + c4)
+            // so that a later rule can pick it up
+            expr = mutate(a - Min::make(min_b->b, min_b->a));
+        } else if (no_overflow(op->type) &&
+                   max_a &&
+                   max_b &&
+                   is_simple_const(mutate(max_a->a - max_b->b)) &&
+                   is_simple_const(mutate(max_a->b - max_b->a))) {
+            // Canonicalize max(a + c1, b + c2) - max(b + c4, a + c3)
+            //     where c1, c2, c3, and c4 are constants
+            // into max(a + c1, b + c2) - max(a + c3, b + c4)
+            // so that a later rule can pick it up
+            expr = mutate(a - Max::make(max_b->b, max_b->a));
+        } else if (no_overflow(op->type) &&
+                   min_a &&
+                   min_b) {
+            // min(a + c1, b + c2) - min(a + c3, b + c4)
+            //     where delta_a = c1 - c3 and delta_b = c2 - c4 are constants
+            // if delta_b - delta_a <= 0 -> clamp((b + c2) - (a + c1), delta_b - delta_a, 0) + delta_a
+            // else -> delta_b - clamp((b + c2) - (a + c1), 0, delta_b - delta_a)
+            Expr delta_a = mutate(min_a->a - min_b->a);
+            Expr delta_b = mutate(min_a->b - min_b->b);
+            if (is_simple_const(delta_a) &&
+                is_simple_const(delta_b)) {
+                Expr diff = delta_b - delta_a;
+                if (is_one(mutate(diff <= 0))) {
```

Comment:
Elsewhere we use make_zero(op->type) instead of relying on the literal zero to do the right thing. I can't recall why, unfortunately. Might be safest to make these make_zero calls though. Things like diff could be constant vectors.